Compilation error `error[E0277]`

I tried to use this library for using my vl53loc-v2 senor. But when I try to compile the project I get an compilation error for this library:
```
$ cargo build
   Compiling vl53l0x v1.0.0
error[E0277]: the trait bound `u16: cast::From<_>` is not satisfied
   --> /home/mobergmann/.cargo/registry/src/index.crates.io-6f17d22bba15001f/vl53l0x-1.0.0/src/lib.rs:155:17
    |
155 |         Ok((u16(buffer[0]) << 8) + u16(buffer[1]))
    |             --- ^^^^^^^^^ the trait `cast::From<_>` is not implemented for `u16`
    |             |
    |             required by a bound introduced by this call
    |
    = help: the trait `cast::From<f64>` is implemented for `f32`
    = help: for that trait implementation, expected `f32`, found `u16`
note: required by a bound in `u16`
   --> /home/mobergmann/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cast-0.2.7/src/lib.rs:181:1
    |
181 | fns!(f32, f64, i8, i16, i32, i64, isize, u8, u16, u32, u64, usize);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^---^^^^^^^^^^^^^^^^^^
    | |                                            |
    | |                                            required by a bound in this function
    | required by this bound in `u16`
    = note: this error originates in the macro `fns` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `u16: cast::From<_>` is not satisfied
   --> /home/mobergmann/.cargo/registry/src/index.crates.io-6f17d22bba15001f/vl53l0x-1.0.0/src/lib.rs:155:13
    |
155 |         Ok((u16(buffer[0]) << 8) + u16(buffer[1]))
    |             ^^^^^^^^^^^^^^ the trait `cast::From<_>` is not implemented for `u16`
    |
    = help: the trait `cast::From<f64>` is implemented for `f32`
    = help: for that trait implementation, expected `f32`, found `u16`

error[E0277]: the trait bound `u16: cast::From<_>` is not satisfied
   --> /home/mobergmann/.cargo/registry/src/index.crates.io-6f17d22bba15001f/vl53l0x-1.0.0/src/lib.rs:155:40
    |
155 |         Ok((u16(buffer[0]) << 8) + u16(buffer[1]))
    |                                    --- ^^^^^^^^^ the trait `cast::From<_>` is not implemented for `u16`
    |                                    |
    |                                    required by a bound introduced by this call
    |
    = help: the trait `cast::From<f64>` is implemented for `f32`
    = help: for that trait implementation, expected `f32`, found `u16`
note: required by a bound in `u16`
   --> /home/mobergmann/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cast-0.2.7/src/lib.rs:181:1
    |
181 | fns!(f32, f64, i8, i16, i32, i64, isize, u8, u16, u32, u64, usize);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^---^^^^^^^^^^^^^^^^^^
    | |                                            |
    | |                                            required by a bound in this function
    | required by this bound in `u16`
    = note: this error originates in the macro `fns` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `u16: cast::From<_>` is not satisfied
   --> /home/mobergmann/.cargo/registry/src/index.crates.io-6f17d22bba15001f/vl53l0x-1.0.0/src/lib.rs:155:36
    |
155 |         Ok((u16(buffer[0]) << 8) + u16(buffer[1]))
    |                                    ^^^^^^^^^^^^^^ the trait `cast::From<_>` is not implemented for `u16`
    |
    = help: the trait `cast::From<f64>` is implemented for `f32`
    = help: for that trait implementation, expected `f32`, found `u16`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `vl53l0x` (lib) due to 4 previous errors
```

My project is basically straight out of the cargo generate tool for the Arduino Uno. I customized it a bit, but basically it looks like this:
```rust
    let mut tof = vl53l0x::VL53L0x::new(i2c).expect("tof");
    tof.set_measurement_timing_budget(200000).expect("time budget");
    tof.start_continuous(0).expect("start");

    // let duration = Milliseconds::<u64>(embedded_time::duration::ms);

    loop {
        led.toggle();
        
        let mls = tof.read_range_continuous_millimeters();
        ufmt::uwriteln!(&mut serial, "Distance: {:?}?", mls).unwrap();
    }
```

I am using nixos with a rustup shell as according to the [nixos wiki](https://nixos.wiki/wiki/Rust#Installation_via_rustup):
```
[nix-shell:~/Source/vl53l0x]$ cargo --version
cargo 1.77.1

[nix-shell:~/Source/vl53l0x]$ rustc --version
rustc 1.77.2 (25ef9e3d8 2024-04-09) (built from a source tarball)
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,10 +13,6 @@ edition = "2018"
 [dependencies]
 nb = "0.1"
 
-[dependencies.cast]
-default-features = false
-version = "0.2.2"
-
 [dependencies.hal]
 version = "1.0"
 package = "embedded-hal"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -5,7 +5,6 @@
 #![allow(dead_code)]
 #![no_std]
 
-use cast::u16;
 use hal::i2c::I2c;
 
 const DEFAULT_ADDRESS: u8 = 0x29;
@@ -152,7 +151,7 @@ where
     fn read_16bit(&mut self, reg: Register) -> Result<u16, E> {
         let mut buffer: [u8; 2] = [0, 0];
         self.read_registers(reg, &mut buffer)?;
-        Ok((u16(buffer[0]) << 8) + u16(buffer[1]))
+        Ok(((buffer[0] as u16) << 8) + (buffer[1] as u16))
     }
 
     fn write_byte(&mut self, reg: u8, byte: u8) -> Result<(), E> {

